### PR TITLE
Close button to go back to recipe cards page

### LINKED ISSUE #39: Add ability to go back from recipeDetails
**Is your feature request related to a problem? Please describe.**
Users have no ability to go back when viewing recipe details other than refreshing the page.

**Describe the solution you'd like**
Add a button to allow user to navigate back through previously viewed information.

### PR DESCRIPTION
**Pull request by connielion**
## Changes
1. Defined function to reset boolean hooks for conditionally rendering RecipeCards page from RecipeDetails.
2. Removed `setSelectedIngredients([])` line in search function to keep ingredients list in Sidebar.
3. Pulled from development and refactored `addRecipe` function in MainContainer to not allow same recipe to be added more than once.

## Purpose
Conditional render RecipeCards from clicking close button in RecipeDetails.

## Approach
React hooks/typescript.

Closes #39

### CODE REVIEW

File: src/components/Sidebar.tsx [hunk 1]
```diff
@@ -81,7 +81,7 @@ const Sidebar: FC<SidebarProps> = ({
       setUserSearchedRecipes(true);
       const recipesList = await searchByIngredients(selectedIngredients.join(','));
       setSearchedRecipes(recipesList);
-      setSelectedIngredients([]);
+      // setSelectedIngredients([]);
```
Comment (Lap343) [inline src/components/Sidebar.tsx:84]: is this still needed?

Comment (BlackCubes) [inline src/components/Sidebar.tsx:84]: Do ye need dis comment?

**Review by BlackCubes — APPROVED** [review]
Looks gewd! Just remove comment if ye don't need it

File: src/components/MainContainer.tsx [hunk 2]
```diff
@@ -98,8 +98,16 @@ const MainContainer = () => {
         ingredient,
       ]);
     }
+  }
+
+  const addRecipe = (e: React.MouseEvent<HTMLDivElement>, recipe: IRecipe) => {
+    e.stopPropagation();
+    if(!selectedRecipes.includes(recipe)){
+      setSelectedRecipes((prevState: IRecipe[]) => [...prevState, recipe]);
+    }
```
Comment (krferrell) [inline src/components/MainContainer.tsx:107]: Nice work here. Having to use the React events was interesting to me.

**Review by krferrell — APPROVED** [review]
Looks good! Keep up the good work.

**Pull request merged**